lib/interactivity-api.php

```diff
@@ -24,9 +24,31 @@ function gutenberg_reregister_interactivity_script_modules() {
 	wp_register_script_module(
 		'@wordpress/interactivity-router',
 		gutenberg_url( '/build-module/interactivity-router/index.min.js' ),
-		array( '@wordpress/interactivity' ),
+		array(
+			array(
+				'id'     => '@wordpress/a11y',
+				'import' => 'dynamic',
+			),
+			'@wordpress/interactivity',
+		),
 		$default_version
 	);
 }
-
 add_action( 'init', 'gutenberg_reregister_interactivity_script_modules' );
+
+function gutenberg_register_interactivity_script_module_data_hooks() {
```

Comment:
What is the reason `gutenberg_reregister_interactivity_script_modules` and `gutenberg_register_interactivity_script_module_data_hooks` are separated? It seems like it should be enough to deregister `@wordpress/interactivity-router` and override the config for the script module in one callback.

Reply:
[Core registers the Interactivity (and script module) hooks on `after_setup_theme`.](https://github.com/WordPress/wordpress-develop/blob/d2ce8ddbe026ce33115c1bb6e60d3efbb96fa6ed/src/wp-settings.php#L405) If the filter is registered earlier, the conditional filter registration won't be conditional because the Core filter will not have been registered yet.

This is the same approach taken with the script module data compatibility in Gutenberg in https://github.com/WordPress/gutenberg/pull/64006:

https://github.com/WordPress/gutenberg/blob/4ca78cdf8ec71ba62dbf4403662202872d8e5888/lib/compat/wordpress-6.7/script-modules.php#L92-L104

Reply:
I need to investigate it further because it’s unclear why configuring the script module would require a different hook than it’s registration. In both cases we need to ensure that core does it’s work fist and then the plugin overrides everything.

Reply:
It's all consolidated in WordPress core in one method for Interactivity API with two levels of hooks, so it's difficult to reason about:


https://github.com/WordPress/wordpress-develop/blob/d2ce8ddbe026ce33115c1bb6e60d3efbb96fa6ed/src/wp-includes/interactivity-api/class-wp-interactivity-api.php#L286-L312

Let's leave it for later.